src/WorkBC.Shared/Services/GeocodingCachingService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using WorkBC.Data;
+using WorkBC.Data.Model.JobBoard;
+
+namespace WorkBC.Shared.Services
+{
+    public class GeocodingCachingService : IGeocodingService
+    {
+        private readonly JobBoardContext _dbContext;
+        private readonly ILogger<IGeocodingService> _logger;
+        private readonly IGeocodingService _geocodingService;
+
+        public GeocodingCachingService(JobBoardContext dbContext, IGeocodingService geocodingService,
+            ILogger<IGeocodingService> logger = null)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+            _geocodingService = geocodingService;
+        }
+
+        public async Task<GeocodedLocationCache> GetLocation(string location)
+        {
+            GeocodedLocationCache cachedLocation =
+                await _dbContext.GeocodedLocationCache.FirstOrDefaultAsync(g => g.Name == location);
+
+            if (cachedLocation != null)
+            {
+                return cachedLocation;
+            }
+
+            // no cache hit; call the geocoding service and save to cache
+            var locationObject = await _geocodingService.GetLocation(location);
+            return await SetLocation(locationObject);
+        }
+
+        public async Task<GeocodedLocationCache> SetLocation(GeocodedLocationCache geo)
+        {
+            if (geo.Latitude != null && geo.Longitude != null)
+            {
+                try
+                {
+                    _dbContext.GeocodedLocationCache.Add(geo);
+                    await _dbContext.SaveChangesAsync();
+                    return geo;
+                }
+                catch (DbUpdateException e)
+                {
+                    _logger?.LogError("DBUpdateException: " + e.Message);
+                    throw;
+                }
+            }
+            return null;
+        }
+        
+        public async Task<bool> DeleteLocation(GeocodedLocationCache geo)
+        {
+            GeocodedLocationCache cachedLocation =
+                await _dbContext.GeocodedLocationCache.FirstOrDefaultAsync(g => g.Name == geo.Name);
```

Comment:
I'd suggest wrapping the comparison function into a static method, to ensure we can edit it reliably.

Reply:
I like that idea!  Standby fix in the works ...